armi/reactor/blocks.py

```diff
@@ -2379,7 +2381,18 @@ def autoCreateSpatialGrids(self, cornersUp=True):
 
         # build the grid, from pitch and orientation
         grid = grids.HexGrid.fromPitch(
-            self.getPinPitch(cold=True), numRings=0, cornersUp=cornersUp
+            self.getPinPitch(cold=True),
+            numRings=0,
+            armiObject=self,
+            cornersUp=cornersUp,
+        )
+
+        # build the sub-grid, with opposite orientation to the block
+        subGrid = grids.HexGrid.fromPitch(
+            self.getPinPitch(cold=True),
+            numRings=0,
+            armiObject=self,
+            cornersUp=not cornersUp,
```

Comment:
This is a deviation from how ARMI has previously reported the pin locations. On main, the following takes place
```python
>>> r = armi.init(fName="armi/tests/tutorials/anl-afci-177.yaml").r
>>> a = r.core.getFirstAssembly(Flags.FUEL)
>>> b = a.getFirstBlock(Flags.FUEL)
>>> fuel = b.getChildrenWithFlags(Flags.FUEL)[0]
>>> fuel.spatialLocator.grid is b.spatialGrid
True
```
The docs for `Composite.spatialGrid` and `Composite.spatialLocator` could be better, but they allude to this fact

https://github.com/terrapower/armi/blob/01b0adee861c3d9ffe9430b709ea1f1db39ca790/armi/reactor/composites.py#L322-L325

So `Composite.spatialGrid` is the sub-grid, or the grid on which the children live. `Core.spatialGrid` is the grid for all the assemblies, `Assembly.spatialGrid` for blocks, `Block.spatialGrid` for components.

I think the negation needs to go up a level higher:
1. `HexBlock.autoCreateSpatialGrid` makes the `HexBlock.spatialGrid` in the orientation prescribed by `cornersUp`
2. All the child components live on the grid defined by `HexBlock.spatialGrid`
3. When we add the `HexAssembly` to a `Core.spatialGrid` (or SFP) we need to invert the orientation with e.g.,
```python
def add(self, hexAssem):
    blockGridCornersUp = not self.spatialGrid.cornersUp
    for b in hexAssem:
        b.autoCreateSpatialGrid(..., cornersUp=blockGridCornersUp)
```